unipop-core/src/org/unipop/structure/UniVertexProperty.java

```diff
@@ -1,17 +1,19 @@
 package org.unipop.structure;
 
+import org.apache.commons.lang.NotImplementedException;
 import org.apache.tinkerpop.gremlin.structure.*;
 import org.apache.tinkerpop.gremlin.structure.util.*;
 
-import java.util.Iterator;
+import java.util.*;
 
-public class UniVertexProperty<V> implements VertexProperty<V> {
+public class UniVertexProperty<V> extends UniElement implements VertexProperty<V> {
```

Comment:
I've told you I accidentally started implementing meta-properties if it doesn't hurt let's keep it that way for future use